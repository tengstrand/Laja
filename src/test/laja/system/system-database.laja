
#connectToDatabase (
   driver: "org.hsqldb.jdbcDriver"
   connection: "jdbc:hsqldb:file:src/test/laja/system/hsqldb/test;ifexists=true"
   user: "sa"
   password: ""
) into con

#con.readMetadata() into $
#con.executeSql("select * from person") into persons
#con.executeSql("select street_name, city from address") into addresses

#write "target/test-output/system-database.out"

Database: {con.metadata.databaseProductName} ({con.metadata.databaseProductVersion})

Tables:
  #foreach (table in tables where table.schema == "public")
    {table.name}
    #foreach (column in table.columns)
       {column.name}
    #end
  #end

Relations:
  #foreach (relation in relations where relation.pkTableSchema == "public")
     {relation.fkTableName}.{relation.fkColumnName} -> {relation.pkTableName}.{relation.pkColumnName}
  #end

Rows in table person:
  #foreach (person in persons)
    {person}
  #end

Rows in table address:
  #foreach (address in addresses)
    {address.street_name}, {address.city}
  #end
#end

#con.close()
