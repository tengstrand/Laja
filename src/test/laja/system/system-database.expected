
Database: HSQL Database Engine (1.8.0)

Tables:
    address
       id
       city
       street_name
    person
       id
       name
       age
       address_id

Relations:
     person.address_id -> address.id

Rows in table person:
    {id=1, name=Adam, age=25, address_id=1}
    {id=2, name=Eve, age=20, address_id=null}

Rows in table address:
    First street, Stockholm
    Second street, Uppsala
